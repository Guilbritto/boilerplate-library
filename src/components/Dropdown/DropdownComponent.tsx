import React, { BaseSyntheticEvent, useState } from 'react'

import { useTheme } from '../../hooks/useTheme'
import {
    Container,
    DropdownContainerItem,
    DropdownItem,
    DropdownTitle
} from './DropdownComponent.styles'
import {
    DropdownProps,
    DropdownItem as DropdownItemType
} from './DropdownComponent.types'
import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/io'

const DropdownComponent = ({
    items = [],
    title,
    multSelect = false,
    selectedItem, 
    setSelectedItem,
}: DropdownProps) => {
    const [open, setOpen] = useState(false)

    const theme = useTheme()

    const handleClick = () => {
        setOpen(!open)
    }
    const handleItemClick = (item: DropdownItemType) => {
        setSelectedItem(item)
        setOpen(false)
    }

    return (
        <Container theme={theme} onClick={handleClick}>
            <DropdownTitle theme={theme}>
                {selectedItem?.value ? selectedItem.value : title}
            </DropdownTitle>
            {open ? <IoIosArrowUp /> : <IoIosArrowDown />}

            <DropdownContainerItem theme={theme} isOpen={open}>
                {items.map(item => (
                    <DropdownItem
                        theme={theme}
                        key={item.id}
                        onClick={() => handleItemClick(item)}
                    >
                        {item.value}
                    </DropdownItem>
                ))}
            </DropdownContainerItem>
        </Container>
    )
}
export default DropdownComponent
