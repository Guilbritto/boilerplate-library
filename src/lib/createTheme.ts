import { defaultTheme } from "../styles/defaultTheme";

export const createTheme = (theme: typeof defaultTheme) => {
  return {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...theme.colors,
    },
    fontSizes: {
      ...defaultTheme.fontSizes,
      ...theme.fontSizes,
    },
  } as typeof defaultTheme;
}