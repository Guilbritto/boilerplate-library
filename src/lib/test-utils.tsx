import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '../styles/defaultTheme'

const Providers: FC = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
