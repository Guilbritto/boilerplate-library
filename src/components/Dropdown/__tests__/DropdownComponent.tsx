import { fireEvent, render } from '../../../lib/test-utils'

import DropdownComponent from '../DropdownComponent'
import { DropdownItem } from '../DropdownComponent.types'

describe('<DropdownComponent />', () => {
    it('should render the component', () => {
        const { container } = render(
            <DropdownComponent
                selectedItem={{} as DropdownItem}
                setSelectedItem={() => {}}
                title="Componente Teste"
                items={[
                    {
                        id: 1,
                        value: 'Testando 123...'
                    }
                ]}
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })
    it('should be able to click on component', () => {
        const { container } = render(
            <DropdownComponent
                selectedItem={{} as DropdownItem}
                setSelectedItem={() => {}}
                title="Componente Teste"
                items={[
                    {
                        id: 1,
                        value: 'Testando 123...'
                    }
                ]}
            />
        )
        fireEvent.click(container.firstChild)
        expect(container.firstChild).toMatchSnapshot()
    })
})
