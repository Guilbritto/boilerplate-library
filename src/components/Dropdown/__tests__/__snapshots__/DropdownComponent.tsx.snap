// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownComponent /> should be able to click on component 1`] = `
<span
  class="sc-eCstlR exfwVj"
/>
`;

exports[`<DropdownComponent /> should render the component 1`] = `
<span
  class="sc-eCstlR exfwVj"
/>
`;
